RedacteurPortaal.Grains/Grains/ArchiveGrain.cs

```diff
@@ -18,37 +20,173 @@ public class ArchiveGrain : Grain, IArchiveGrain
         this.archive = archive;
     }
 
-    public async Task Delete()
+    public Task<ArchiveModel> Get()
     {
-        await this.archive.ClearStateAsync();
+        var state = this.archive.State;
+        return Task.FromResult(state);
+    }
+
+    public async Task<MediaVideoItem> GetVideoItem(Guid guid)
+    {
+        var videoItems = this.archive.State.MediaVideoItems;
+        var videoItem = videoItems.Find(x => x.Id.Equals(guid));
+        return await Task.FromResult(videoItem);
+    }
+
+    public async Task<MediaPhotoItem> GetPhotoItem(Guid guid)
+    {
+        var photoItems = this.archive.State.MediaPhotoItems;
+        var photoItem = photoItems.Find(x => x.Id.Equals(guid));
+        return await Task.FromResult(photoItem);
+    }
+
+    public async Task<MediaAudioItem> GetAudioItem(Guid guid)
+    {
+        var audioItems = this.archive.State.MediaAudioItems;
+        var audioItem = audioItems.Find(x => x.Id.Equals(guid));
+        return await Task.FromResult(audioItem);
+    }
+
+    public async Task<NewsItemModel> GetNewsItem(Guid guid)
+    {
+        var newsItems = this.archive.State.NewsItems;
+        var newsItem = newsItems.Find(x => x.Id.Equals(guid));
+        return await Task.FromResult(newsItem);
+    }
+
+    public async Task<Task<ArchiveModel>> CreateArchive(ArchiveModel archive)
+    {
+        this.archive.State = archive;
+        await this.archive.WriteStateAsync();
+        return Task.FromResult(this.archive.State);
     }
 
-    public async Task AddVideoItem(MediaVideoItem videoItem)
+    public async Task<Task<MediaVideoItem>> AddVideoItem(MediaVideoItem videoItem)
     {
         this.archive.State.MediaVideoItems.Add(videoItem);
         await this.archive.WriteStateAsync();
+        return Task.FromResult(videoItem);
     }
 
-    public async Task AddPhotoItem(MediaPhotoItem photoItem)
+    public async Task<Task<MediaPhotoItem>> AddPhotoItem(MediaPhotoItem photoItem)
     {
         this.archive.State.MediaPhotoItems.Add(photoItem);
         await this.archive.WriteStateAsync();
+        return Task.FromResult(photoItem);
     }
 
-    public async Task AddAudioItem(MediaAudioItem audioItem)
+    public async Task<Task<MediaAudioItem>> AddAudioItem(MediaAudioItem audioItem)
     {
         this.archive.State.MediaAudioItems.Add(audioItem);
         await this.archive.WriteStateAsync();
+        return Task.FromResult(audioItem);
     }
 
-    public async Task Update(ArchiveModel archive)
+    public async Task<Task<NewsItemModel>> AddNewsItem(NewsItemModel newsItem)
```

Comment:
hier ook met die task task. Weet niet of dat werkt

Reply:
ook aangepast